### DISCUSSION TITLE
How to set up the logs?

### DISCUSSION THREAD

**Discussion by antipopp** [drakkan/sftpgo#603] [Q&A] [answered]
I'm trying to find the logs, especially those for failed connection etc. but I'm lost.

I have SFTPGo 2.1.2 running on an EC2 Ubuntu instance, it's installed via apt and I've set `SFTPGO_LOG_FILE_PATH=/etc/sftpgo/sftpgo.log`, but there's nothing there and the `/events` api always returns "Not found". What am I doing wrong?

**Comment by drakkan** [comment, marked as answer]
Hi,

on EC2 you could install sftpgo from the marketplace, this way you will help the project and you have a dedicated support email

https://aws.amazon.com/marketplace/seller-profile?id=6e849ab8-70a6-47de-9a43-13c3fa849335

For the marketplace instances the logs are in `/srv/sftpgo/logs/sftpgo.log`.

To configure a dedicated log file you can use something like this:

```shell
echo "SFTPGO_LOG_FILE_PATH=/srv/sftpgo/logs/sftpgo.log" > /etc/sftpgo/sftpgo.env
chown sftpgo:sftpgo /etc/sftpgo/sftpgo.env
```

The `events` endpoint is available in the development version and will be part of v2.2.0, I suppose you are using v2.1.2, please refer to the docs [here](https://github.com/drakkan/sftpgo/blob/2.1.x/httpd/schema/openapi.yaml)

**Reply by antipopp** [reply]
Thank you, so far I'm just playing with it on my school's AWS so I can't buy the marketplace version. Is the marketplace version the only way to print the logs on EC2?

**Reply by drakkan** [reply]
> Thank you, so far I'm just playing with it on my school's AWS so I can't buy the marketplace version. Is the marketplace version the only way to print the logs on EC2?

The marketplace version is the same SFTPGo version you can download from here it only has some different configurations (one of these is a dedicated log file by default instead of logging to systemd). SFTPGo doesn't have a commercial version with more / different features, I believe in open source development but I can't support you all for free.

Please read my answer above to learn how to configure logging, it is also documented [here](https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md)

**Reply by antipopp** [reply]
Thank you, I had set the env var as a system variable. I'll see if it works this way.

**Reply by kimsonvu** [reply]
Hi,

I deploy docker SFTPGO on Unraid. How should i config with log file? I am newbie.

Thanks you!

**Comment by nomandera** [comment]
Adding some more info for people land here via google like I did. As of today version `SFTPGo 2.7.0` via apt on Debian 13.

```
echo "SFTPGO_LOG_FILE_PATH=/srv/sftpgo/logs/sftpgo.log" > /etc/sftpgo/sftpgo.env
chown sftpgo:sftpgo /etc/sftpgo/sftpgo.env
```
still works but it has to be EXACTLY `/etc/sftpgo/sftpgo.env`. Alternate locations you might be used to such as `/etc/sftpgo/env.d/logging.env` appear not to work for this specific environmental variable.

I am sure there is very good reasoning behind this but it left me quite confused for longer than I would care to admit.
